src/Symfony/Component/Routing/Loader/RouteLoaderInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Defines a class that is able to load and return a RouteCollection.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+interface RouteLoaderInterface
+{
+    /**
+     * @param LoaderInterface $loader
+     *
+     * @return RouteCollection
+     */
+    public function getRouteCollection(LoaderInterface $loader);
```

Comment:
I've type-hinted the interface, but in reality, I'm passing in the concrete `ServiceRouteLoader` itself. The difference is that `ServiceRouteLoader` (like all loaders), has a really handy `import()` method, while `LoaderInterface` does not. I'd prefer to type-hint something with the `import()` method - either the concrete `Loader` or a new interface.